lib/rpmsg/rpmsg_virtio.c

```diff
@@ -478,10 +478,11 @@ static int rpmsg_virtio_ns_callback(struct rpmsg_endpoint *ept, void *data,
 	_ept = rpmsg_get_endpoint(rdev, name, RPMSG_ADDR_ANY, dest);
 
 	if (ns_msg->flags & RPMSG_NS_DESTROY) {
-		if (_ept) {
-			metal_mutex_release(&rdev->lock);
-			rpmsg_destroy_ept(_ept);
-		}
+		if (_ept)
+			_ept->dest_addr = RPMSG_ADDR_ANY;
+		metal_mutex_release(&rdev->lock);
+		if (_ept && _ept->destroy_cb)
+				_ept->destroy_cb(ept);
```

Comment:
the destroyed callback is to used to notify application the endpoint has been destroyed, better to update the name service bind endpoint callback to also handle destroy callback by adding argument to indicate if it is destroy or announcement. but the rpmsg_virtio_ns_callback should still have default behavior if no such callback is defined by application.

Reply:
i will do a patch in this way

Reply:
I reanalyses the point. We have following use cases:
- we receive NS_DESTROY message:  
  => call a unbind call back to inform application that the remote endpoint is destroyed. apllication can take appropriate action (suspend communication for this service, destroy the local endpoint...)
- the local endpoint is destroyed. 
=> only use case is the rpmsg_deinit_vdev call. In this case the rpmsg dev is released. I think there is no reason to call a destroy callback during the deinit; application should ensure that all the communications are stopped before calling the denit . So we could just clean the callback call in  rpmsg_deinit_vdev.

In conclusion, i will propose a patch that :
- rename "destroy_cb" in "unbind_cb"
- suppress the callback call in rpmsg_deinit_vdev.